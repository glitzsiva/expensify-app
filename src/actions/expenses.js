import uuid from 'uuid';



//Add Expenses
export const addExpenses=(
    {
        note='',
        amount=0, 
        description='',
        createdAt=0
    }={}
    )=>({
        type:'ADD_EXPENSES',
        expenses:{
            id:uuid(),
            note,
            amount, 
            description,
            createdAt
        }
});

//Remove Expenses
export const removeExpenses=(
    ({id})=>({
        type:'REMOVE_EXPENSES',
        id
    })
);

// Edit expenses
export const editExpenses=(
    (id,updates)=>({
        type:'EDIT_EXPENSES',
        id,
        updates  
    })
);