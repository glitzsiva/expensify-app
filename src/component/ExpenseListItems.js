import React from 'react';
import {Link} from 'react-router-dom';
import moment from 'moment';
import numeral from 'numeral';

const ExpenseListItems =({description, id,amount,createdAt})=>(
    <div>

        <Link to={`/edit/${id}`}>
            <h1>{description}</h1>
        </Link>
        <h5>
        {numeral(amount).format('$0,0.00')}
        -
        {moment(createdAt).format('MMMM Do,YYYY')}</h5>
        
    </div>
)
export default ExpenseListItems;