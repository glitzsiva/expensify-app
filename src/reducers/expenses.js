


//expenses Reducers DefaultState

const expensesReducersDefaultState=[];

//expenses Reducers

export default (state=expensesReducersDefaultState, action)=>{

    switch(action.type){
        case 'ADD_EXPENSES':
            {
                return [
                    ...state,action.expenses
                ]
            }
        case 'REMOVE_EXPENSES':
            {
                return state.filter(({id}) => id !== action.id)
            }
        case 'EDIT_EXPENSES':
            {
                return state.map((expenses)=> {
                    if (expenses.id !== action.id){
                        return expenses;          
                    } 
                    else{
                        return {...expenses,...action.updates}
                    }
                });
            }
        default:
            {
                return state
            }
    }
}