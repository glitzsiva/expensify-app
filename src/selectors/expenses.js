import moment from 'moment';


export default (expenses,{startDate,endDate,text,sortBy})=>{
    return expenses.filter((expense)=>{
        const momentCreatedAt=moment(expense.createdAt);
        const startDateMatch= startDate ? startDate.isSameOrBefore(momentCreatedAt,'day') :true;
        const endDateMatch= endDate ? endDate.isSameOrAfter(momentCreatedAt ,'day'):true
        const textMatch= expense.description.toLowerCase().includes(text.toLowerCase());
        return startDateMatch && endDateMatch && textMatch;
    }).sort((a,b)=>{
        if(sortBy==='date'){
            return a.createdAt < b.createdAt ?-1 :1;
        }
        else if(sortBy==='amount'){
            return a.amount < b.amount ? -1:1;
        }
    })
    
}