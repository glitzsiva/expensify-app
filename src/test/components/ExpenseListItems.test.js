import React from 'react';
import {shallow} from 'enzyme';
import ExpenseListItems from '../../component/ExpenseListItems';
import expenses from '../fixtures/expenses';


test('Expenses List Items ',()=>{
    const wrapper = shallow( <ExpenseListItems {...expenses[1]} /> );
    expect(wrapper).toMatchSnapshot();
})