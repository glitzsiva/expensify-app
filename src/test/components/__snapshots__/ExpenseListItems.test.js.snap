// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expenses List Items  1`] = `
<div>
  <Link
    replace={false}
    to="/edit/12"
  >
    <h1>
      rents
    </h1>
  </Link>
  <h5>
    $10,000.10
    -
    January 5th,1970
  </h5>
</div>
`;
